code/modules/cm_aliens/hivebuffs/hivebuff.dm

```diff
@@ -0,0 +1,525 @@
+	return TRUE
+
+/datum/hivebuff/proc/_seek_queen_approval(mob/living/purchasing_mob)
+	if(!hive.living_xeno_queen)
+		return FALSE
+
+	var/mob/living/queen = hive.living_xeno_queen
+	var/queen_response = tgui_alert(queen, "You are trying to Purchase [name] at a cost of [cost] [BUFF_POINTS_NAME]. Our hive has [hive.buff_points] [BUFF_POINTS_NAME]. Are you sure you want to purchase it? Description: [desc]", "Approve Hive Buff", list("Yes", "No"), 20 SECONDS)
+
+	return queen_response == "Yes"
+
+/// Any effects which need to be ended or ceased gracefully, called when a buff expires.
+/datum/hivebuff/proc/on_cease()
+	return
+
+/datum/hivebuff/proc/_announce_buff_engage()
+	if(engage_flavourmessage)
+		xeno_announcement(SPAN_XENOANNOUNCE(engage_flavourmessage), hive.hivenumber, XENO_GENERAL_ANNOUNCE)
+
+	if(marine_flavourmessage)
+		marine_announcement(marine_flavourmessage, COMMAND_ANNOUNCE, 'sound/AI/bioscan.ogg')
+
+/datum/hivebuff/proc/_announce_buff_cease()
+	if(!duration)
+		return
+
+	for(var/mob/living/xenomorph as anything in hive.totalXenos)
+		if(!xenomorph.client)
+			continue
+		xenomorph.play_screen_text(cease_flavourmessage, override_color = "#740064")
+		to_chat(xenomorph, SPAN_XENO(cease_flavourmessage))
+
+///Signal handler for new xenomorphs joining the hive
+/datum/hivebuff/proc/_handle_xenomorph_new(datum/source, mob/living/carbon/xenomorph/new_xeno)
+	SIGNAL_HANDLER
+	if(!apply_on_new_xeno)
+		return
+
+	if(!(src in hive.active_hivebuffs))
+		return
+	// If we're the same hive as the buff
+	if(new_xeno.hive == hive)
+		apply_buff_effects(new_xeno)
+
+///The actual effects of the buff to apply
+/datum/hivebuff/proc/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	return
+
+/// Reverse the effects here, should be the opposite of apply_effects()
+/datum/hivebuff/proc/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	return
+
+/datum/hivebuff/proc/handle_special_checks()
+	return TRUE
+
+
+// BUFFS //
+
+/datum/hivebuff/extra_larva
+	name = "Surge of Larva"
+	desc = "Provides 5 larva instantly to the hive."
+	radial_icon = "larba"
+
+	engage_flavourmessage = "The Queen has purchased 5 extra larva to join the hive!"
+	cost = 5
+	number_of_required_pylons = 1
+	is_reusable = FALSE
+
+/datum/hivebuff/extra_larva/on_engage(obj/effect/alien/resin/special/pylon/purchased_pylon)
+	hive.stored_larva += 5
+	return TRUE
+
+/datum/hivebuff/evo_buff
+	name = "Boon of Evolution"
+	desc = "Doubles evolution speed for 5 minutes."
+	tier = HIVEBUFF_TIER_MINOR
+	engage_flavourmessage = "The Queen has blessed us with faster evolution."
+	duration = 5 MINUTES
+	number_of_required_pylons = 1
+	var/value_before_buff
+
+/datum/hivebuff/evo_buff/on_engage(obj/effect/alien/resin/special/pylon/purchased_pylon)
+	var/value_before_buff = hive.evolution_bonus
+	hive.override_evilution(value_before_buff * 2, TRUE)
+
+	return TRUE
+
+/datum/hivebuff/evo_buff/on_cease()
+	hive.override_evilution(value_before_buff, FALSE)
```

Comment:
var/ creates new variable to store value, so on_cease sets evilution to null

Reply:
true